Index: src/quickassist/utilities/osal/include/Osal.h
===================================================================
--- src.orig/quickassist/utilities/osal/include/Osal.h
+++ src/quickassist/utilities/osal/include/Osal.h
@@ -653,7 +653,7 @@ osalMemSet (void *ptr, UINT8 filler, UIN
  *  Time
  *****************************/
 
-int __init osal_init(void);
+int osal_init(void);
 void osal_exit(void);
 
 /* Retrieve current system time */
