--- a/quickassist/adf/drivers/common/linux/src/adf_dev_csr.c
+++ b/quickassist/adf/drivers/common/linux/src/adf_dev_csr.c
@@ -411,7 +411,7 @@
  * VM_RESERVED was removed from the 3.10 kernel and replaced with
  * (VM_DONTEXPAND | VM_DONTDUMP) 
  */
-        vma->vm_flags |= VM_IO | (VM_DONTEXPAND | VM_DONTDUMP);
+        vm_flags_set(vma, VM_IO | (VM_DONTEXPAND | VM_DONTDUMP));
 #endif
 	vma->vm_ops = &adf_mmap_operation;
         vma->vm_page_prot = pgprot_noncached(vma->vm_page_prot);
