Index: src/quickassist/adf/accel_mgr/src/adf_init.c
===================================================================
--- src.orig/quickassist/adf/accel_mgr/src/adf_init.c
+++ src/quickassist/adf/accel_mgr/src/adf_init.c
@@ -435,7 +435,7 @@ icp_adf_subsystemUnregister(subservice_r
  * adf_cfgAddInternalValues
  * Add internal config values to general section of the given accel dev
  */
-CpaStatus
+static CpaStatus
 adf_cfgAddInternalValues(icp_accel_dev_t *accel_dev)
 {
     CpaStatus status = CPA_STATUS_FAIL;
