--- a/quickassist/build_system/build_files/env_files/linux_2.6_user_space.mk
+++ b/quickassist/build_system/build_files/env_files/linux_2.6_user_space.mk
@@ -76,7 +76,8 @@
 INCLUDES+= \
           -I$(API_DIR)   \
           -I$(OSAL_DIR)/include \
-          -I$(OSAL_DIR)/src/linux/user_space/include
+          -I$(OSAL_DIR)/src/linux/user_space/include \
+          -I$(OSAL_DIR)/src/linux/user_space/osalzlib/include

 #----------------------------------------------------------
 #
